PlaywrigthUITests/PageObjects/DemoQAWebTablesPage.cs

```diff
@@ -87,18 +93,124 @@ public async Task VerifyTableRowContent(string headerName = "First Name", string
+    {
+        await AddButton.ClickAsync();
+        Assert.That(await RegistrationPopup.IsVisibleAsync(), "The 'Add' popup is not visible.");
+    }
+    
+    public async Task FillFirstName (string firstName)
+    {
+        if(firstName == null) throw new ArgumentException("First Name field should not be null.");
+        await FirstNameField.FillAsync(firstName);
+        Assert.That(await FirstNameField.GetAttributeAsync("value") == firstName, $"The First Name field was not filled with: '{firstName}'.");
+    }
+    
+    public async Task FillLastName (string lastName)
+    {
+        if(lastName == null) throw new ArgumentException("Last Name field should not be null.");
+        await LastNameField.FillAsync(lastName);
+        Assert.That(await LastNameField.GetAttributeAsync("value") == lastName, $"The Last Name field was not filled with: '{lastName}'.");
+    } 
+    
+    public async Task FillEmail (string email)
+    {
+        if(email == null) throw new ArgumentException("Email field should not be null.");
+        await EmailField.FillAsync(email);
+        Assert.That(await EmailField.GetAttributeAsync("value") == email, $"The Email field was not filled with: '{email}'.");
+    }
+    
+    public async Task FillAge (string age)
+    {
+        if(age == null) throw new ArgumentException("Age field should not be null.");
+        await AgeField.FillAsync(age);
+        Assert.That(await AgeField.GetAttributeAsync("value") == age, $"The Age field was not filled with: '{age}'.");
+    }
+    
+    public async Task FillSalary (string salary)
+    {
+        if(salary == null) throw new ArgumentException("Salary field should not be null.");
+        await SalaryField.FillAsync(salary);
+        Assert.That(await SalaryField.GetAttributeAsync("value") == salary, $"The Salary field was not filled with: '{salary}'.");
+    }
+    
+    public async Task FillDepartment (string department)
+    {
+        if(department == null) throw new ArgumentException("Department field should not be null.");
+        await DepartmentField.FillAsync(department);
+        Assert.That(await DepartmentField.GetAttributeAsync("value") == department, $"The Department field was not filled with: '{department}'.");
+    }
 
-}
+    public async Task ClickSubmit()
+    {
+        var submit = Page.GetByRole(AriaRole.Button, new() { Name = "Submit" });
+        await submit.ClickAsync();
+    }
+
+    public async Task<bool> FindRowInTable(string firstName, string lastName, string email, string age, string salary, string department)
+    {
+        if (string.IsNullOrWhiteSpace(firstName) ||
+            string.IsNullOrWhiteSpace(lastName) ||
+            string.IsNullOrWhiteSpace(email) ||
+            string.IsNullOrWhiteSpace(age) ||
+            string.IsNullOrWhiteSpace(salary) ||
+            string.IsNullOrWhiteSpace(department))
+        {
+            throw new ArgumentException("One or more fields are empty.");
```

Comment:
bad argument exception handling. In case of failure you have to debug what exactly parameter was null. What is the point to check parameters in this way? You can just skip checking, it will fail later and you will debug anyway

Reply:
fixed